src/renderer/features/matrix/MatrixAutoLogin/model/matrix-autologin-model.ts

```diff
@@ -25,68 +26,90 @@ const loginFromCacheFx = createEffect((matrix: ISecureMessenger): Promise<void>
 });
 
 sample({
-  clock: loginStarted,
-  fn: () => AutoLoginStatus.IN_PROCESS,
-  target: $autoLoginStatus,
-});
-
-split({
-  source: loginStarted,
-  match: {
-    withToken: (value) => Boolean(value),
-  },
-  cases: {
-    withToken: loggedInWithToken,
-    __: loggedInFromCache,
+  clock: loggedInFromCache,
+  source: {
+    matrix: matrixModel.$matrix,
+    loginStatus: matrixModel.$loginStatus,
+    autoLoginStatus: $autoLoginStatus,
   },
+  filter: ({ loginStatus }) => matrixUtils.isLoggedOut(loginStatus),
+  fn: ({ matrix }) => ({
+    loginEffect: matrix,
+    loginStatus: LoginStatus.AUTO_LOGIN,
+  }),
+  target: spread({
+    loginEffect: loginFromCacheFx,
+    loginStatus: matrixModel.$loginStatus,
+  }),
 });
 
 sample({
-  clock: [loggedInWithToken, loggedInFromCache],
-  fn: () => LoginStatus.AUTO_LOGIN,
-  target: matrixModel.events.loginStatusChanged,
+  clock: loggedInWithToken,
+  filter: loginFromCacheFx.pending,
```

Comment:
new use case for filter, I will remember it 👍

Reply:
Yeah, someone told me about this in effector chat